lisp/ess-r-package.el

```diff
@@ -87,6 +87,11 @@ whether the current file is part of a package, or the value of
       (with-ess-process-buffer t
         ess-r-package-info)))
 
+(defun ess-r-package-project (dir)
```

Comment:
Why do we need so many functions that do similar stuff `ess-r-package-project`, `ess-r-package--package-dir-info`, `ess-r-package-info`, `ess-r-package--find-package-path`. It's confusing. Cannot we reduce them to 1 max 2 functions?

BTW, shouldn't it be `ess-r-package-project-root` or `...-dir`?

Reply:
There's a function that does the searching + caching, one that returns the package info in ESS format, one that returns the package info in Emacs project format, ... I'm not sure we can streamline it further but a better naming scheme would make things clearer. I'll try to simplify.

> shouldn't it be ess-r-package-project-root

I don't think so because it doesn't return a project root, it returns a cons cell that includes the project name.

Reply:
>  I'll try to simplify.

Thanks!